### PR TITLE
[Transaction] Fix transaction system topic create in loop.

### PR DESCRIPTION
**Pull request by congbobo184**
fix https://github.com/apache/pulsar/issues/12727
### Motivation
Now transaction system topic can be created.

### Modifications
we should not allow broker or user create by transaction system format topic.

1. checkout topic auto create.
2. admin create topic.

### Verifying this change

add some test for it

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 1]
```diff
@@ -1649,14 +1649,21 @@ public void shutdownNow() {
...
         String topic = topicName.toString();
         return topic.startsWith(TopicName.TRANSACTION_COORDINATOR_ASSIGN.toString())
                 || topic.startsWith(TopicName.get(TopicDomain.persistent.value(),
                 NamespaceName.SYSTEM_NAMESPACE, TRANSACTION_LOG_PREFIX).toString())
                 || topic.endsWith(MLPendingAckStore.PENDING_ACK_STORE_SUFFIX);
     }
 
+    public static boolean isNotAllowedToCreateTopic(TopicName topicName) {
+        String topic = topicName.toString();
+        return topic.startsWith(TopicName.get(TopicDomain.persistent.value(),
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:1662]: what about creating a constant for this ? "TopicName.get(TopicDomain.persistent.value(), NamespaceName.SYSTEM_NAMESPACE, TRANSACTION_LOG_PREFIX).toString()"

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -244,6 +245,13 @@ protected void validateAdminAndClientPermission() {
         }
     }
 
+    protected void validateTopicAllowdToCreate(TopicName topicName) {
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:248]: typo: validateCreateTopic

**Comment by eolivelli** [comment]
Can you add my reproducer as a test case ?

I am not sure that we will really fix the problem. Because we are not forbidding to create a topic with a system name.
But in my case the topic creation is triggered by getSubscriptions().
We do not want getSubscriptions() to fail

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 3]
```diff
@@ -1649,11 +1648,16 @@ public void shutdownNow() {
...
+    public static boolean isTransactionSystemTopic(TopicName topicName) {
         String topic = topicName.toString();
         return topic.startsWith(TopicName.TRANSACTION_COORDINATOR_ASSIGN.toString())
-                || topic.startsWith(TopicName.get(TopicDomain.persistent.value(),
-                NamespaceName.SYSTEM_NAMESPACE, TRANSACTION_LOG_PREFIX).toString())
+                || topic.startsWith(TRANSACTION_COORDINATOR_LOG.toString())
+                || topic.endsWith(MLPendingAckStore.PENDING_ACK_STORE_SUFFIX);
+    }
+
+    public static boolean isNotAllowedToCreateTopic(TopicName topicName) {
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:1658]: what about naming this method `isTransactionInternalName` ?
the meaning is that this is a reserved name

I see that you are using this method while listing the topic names, so "Create" is not good in that place

**Review by eolivelli — CHANGES_REQUESTED** [review]
I left one last minor comment.

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by codelipenghui — APPROVED** [review]

**Comment by Anonymitaet** [comment]
Confirmed w/ @codelipenghui, this PR does not need to update docs.

**Pull request merged**